zad20.py

```diff
@@ -0,0 +1,64 @@
+def zad18(tab, start=None, dest=None):
+    n = len(tab)
+    route = []
+
+    if start is None:
+        start = (0, 0)
+    if dest is None:
+        dest = (n - 1, n - 1)
+
+    def first_digit(num):
+        if num == 0:
+            return 0
+        a = 0
+        while num > 0:
+            a = num
+            num //= 10
+        return a
+
+    def last_digit(num):
+        return num % 10
+
+    def steps(pos):
+        _steps = [(pos[0] - 1, pos[1] - 1, 7), (pos[0], pos[1] - 1, 0), (pos[0] + 1, pos[1] - 1, 1),
+                  (pos[0] - 1, pos[1], 6), (pos[0] + 1, pos[1], 2),
+                  (pos[0] - 1, pos[1] + 1, 5), (pos[0], pos[1] + 1, 4), (pos[0] + 1, pos[1] + 1, 3)]
+        return _steps
+
+    def can_move(from_where, where_to):
+        if n > where_to[0] >= 0 and n > where_to[1] >= 0:
+            if abs(dest[0] - from_where[0]) >= abs(dest[0] - where_to[0]) and \
+                    abs(dest[1] - from_where[1]) >= abs(dest[1] - where_to[1]):
+                if last_digit(tab[from_where[0]][from_where[1]]) < first_digit(tab[where_to[0]][where_to[1]]):
+                    return True
+        return False
```

Comment:
Niby można dać od razu `return a and b and c and d` itd ale no cóż.

Reply:
Nie rozumiem za bardzo co masz na myśli.

Reply:
Masz
<code>
if coś and coś:

    if coś and coś:

        return True

return False
</code>

Czyli to samo co `return coś and coś and coś and coś`

Reply:
Faktycznie, ale chyba w jednej linijce by się to zasyfiło. Niby do zmiennych można te wartości i na końcu wrzucić.